Added location state in tracing card screen (EXPOSUREAPP-1757)

Oliver-Zimmerman opened the pull request:
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [ ] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [ ] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) - Pull Requests that are not linked to an issue with you as an assignee will be closed, except for minor fixes for typos or grammar mistakes in *documentation or code comments*.
* [x] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)
* [ ] Make sure that your PR does not contain changes in text files, therefore the PR must not contain changes in values/strings/* and / or assets/* (see issue #332 for further information)
* [x] Make sure that your PR does not contain compiled sources (already set by the default .gitignore) and / or binary files

## Description
- Edited FormatterSettingsHelper.kt to include a visibility formatter based on location state.
- Edited SettingsTracingFragment.kt included a listener for location settings button.
- Added illustration for location missing in tracing screen.
- Edited fragment_settings_tracing.xml so that if location isn't on, a card explaining it should be turned on appears.
- Edited strings values to include required strings for card.

Completes JIRA task EXPOSUREAPP-1757

Oliver-Zimmerman committed:
Implemented state check for location

Oliver-Zimmerman committed:
formatting

Oliver-Zimmerman committed:
removed duplicate manifest item

Oliver-Zimmerman committed:
Update ConnectivityHelper.kt

Oliver-Zimmerman committed:
Update ConnectivityHelper.kt

Oliver-Zimmerman committed:
Merge branch 'dev' into feature/location-state-implementation

Oliver-Zimmerman committed:
Added location type to TracingStatusHelper

Added location type to TracingStatusHelper as well as relevant String values

Oliver-Zimmerman committed:
Merge branch 'dev' into feature/homescreen-location-warning

Oliver-Zimmerman committed:
added location paramater to TracingStatusHelper utilizing methods

added location paramater to TracingStatusHelper utilizing methods.

Implementation including location not yet included.

Oliver-Zimmerman committed:
Finalized TracingStatusHelper to include the location value

Finalized TracingStatusHelper to include the location value to be returned to Formatters in FormatterSettingsHelper

Oliver-Zimmerman committed:
formatting

Oliver-Zimmerman committed:
Update FormatterSettingsHelper.kt

Oliver-Zimmerman committed:
Update FormatterSettingsHelper.kt

Oliver-Zimmerman committed:
Added home screen implementation for location state

- Added logic to formatters adjust home screen tracing state based on location setting.
- Adjusted existing state tests to include location boolean value and results.
- Added location inactive icon.

Oliver-Zimmerman committed:
Update FormatterSettingsHelper.kt

Oliver-Zimmerman committed:
test changes to match formatter/helper changes

Oliver-Zimmerman committed:
Merge branch 'dev' into feature/homescreen-location-warning

Oliver-Zimmerman committed:
Create ic_illustration_location_off.xml

Oliver-Zimmerman committed:
Added location state in tracing card screen

- Edited FormatterSettingsHelper.kt to include a visibility formatter based on location state.
- Edited SettingsTracingFragment.kt included a listener for location settings button.
- Added illustration for location missing in tracing screen.
- Edited fragment_settings_tracing.xml so that if location isn't on, a card explaining it should be turned on appears.
- Edited strings values to include required strings for card.

Oliver-Zimmerman committed:
Merge branch 'dev' into feature/risk-screen-location-warning

pwoessner commented on Corona-Warn-App/src/main/res/values-ar/strings.xml:
```diff
@@ -469,6 +475,8 @@
     <!-- XACT: settings(tracing) - describes illustration -->
+    <string name="settings_tracing_location_illustration_description_inactive">""</string>
+    <!-- XACT: settings(tracing) - describes illustration -->
     <string name="settings_tracing_connection_illustration_description_inactive">"قطع شخص الاتصال بالإنترنت على هاتفه الذكي؛ وبالتالي، لم يتم تسجيل تجمُّع مع شخصين آخرين."</string>
```
Hi Oliver, can you remove all language specific changes (ar, en, tr, ru)? Only the german text is the values-de file is needed and the empty placeholder in the value-base file.

Oliver-Zimmerman committed:
Update strings.xml

Oliver-Zimmerman committed:
Updated strings for translation, keeping only confirmed DE and blank base string values

Updated strings for translation, keeping only confirmed DE and blank base string values

pwoessner committed:
Merge branch 'dev' into feature/risk-screen-location-warning

Oliver-Zimmerman committed:
Merge branch 'dev' into feature/risk-screen-location-warning
